File: pset6/sentimental/mario/more/mario.py
# filename: mario.py
""" Display mario's pyramid of n size """

# get positive int from user
while True:
    # prompt user for a number, check if input is valid
    try:
        height = int(input("Height: "))
    except ValueError:
        continue
    if height >= 0 and height <= 23:
        break

for i in range(height):
    # print whitespace
    for j in range((height - i) - 1):
        print(' ', end='')

    # print left side of pyramid
    for j in range(i + 1):
        print('#', end='')

    # add a gap of two spaces
    print('  ', end='')

    # print right side of pyramid
    for j in range(i + 1):
        print('#', end='')

    # new line
    print()

File: pset1/mario/less/mario.c
#include <stdio.h>
#include <cs50.h>

int main(void)
{
    // get positive integer from user; repeat until input passes check
    int height;
    do
    {
        height = get_int("Height: ");
    }
    while (height < 0 || height > 23);
    // display half pyramid
    for (int i = 0; i < height; i++)
    {
        // display spaces
        for (int j = height - (i + 1); j > 0; j--)
        {
            printf(" ");
        }
        // display bricks
        for (int j = 0; j < i + 1; j++)
        {
            printf("#");
        }
        printf("#\n");
    }
}